services/metadata_service/api/run.py

```diff
@@ -136,6 +143,97 @@ async def create_run(self, request):
 
         return await self._async_table.add_run(run_row, fill_heartbeat=client_supports_heartbeats)
 
+    @format_response
+    @handle_exceptions
+    async def mutate_user_tags(self, request):
+        """
+        ---
+        description: mutate user tags
+        tags:
+        - Run
+        parameters:
+        - name: "flow_id"
+          in: "path"
+          description: "flow_id"
+          required: true
+          type: "string"
+        - name: "run_number"
+          in: "path"
+          description: "run_number"
+          required: true
+          type: "string"
+        - name: "body"
+          in: "body"
+          description: "body"
+          required: true
+          schema:
+            type: object
+            properties:
+                tags_to_add:
+                    type: array of string
+                tags_to_remove:
+                    type: array of string
+        produces:
+        - 'text/plain'
+        responses:
+            "200":
+                description: successful operation. Tags updated.  Returns latest user tags
+            "400":
+                description: invalid HTTP Request
+            "405":
+                description: invalid HTTP Method
+        """
+        flow_name = request.match_info.get("flow_id")
+        run_number = request.match_info.get("run_number")
+        body = await read_body(request.content)
+        tags_to_add = body.get("tags_to_add", [])
+        tags_to_remove = body.get("tags_to_remove", [])
+
+        if not isinstance(tags_to_add, list):
+            return DBResponse(response_code=422, body="tags_to_add must be a list")
```

Comment:
It's debatable whether to return a 400 or a 422 here. No strong opinion but 422 is a bit more obscure than 400. As long as the client deals with it ok, I suppose it is fine though.